pkg/ebpf/tracee.go

```diff
@@ -778,7 +778,11 @@ func (t *Tracee) populateBPFMaps() error {
 
 	cZero := uint32(0)
 	configVal := make([]byte, 208)
-	binary.LittleEndian.PutUint32(configVal[0:4], uint32(os.Getpid()))
+	selfPid := 0
+	if t.config.Filter.TraceSelf {
+		selfPid = os.Getpid()
```

Comment:
So the logic [here](https://github.com/aquasecurity/tracee/blob/main/pkg/ebpf/c/tracee.bpf.c#L1822)  is what determines self tracing.
So if we want to enable self tracing we actually need to pass some impossible PID here (-1 or the MAX_PID+1 on the system).
Am I missing something in this assessment?

Reply:
That would be the case if you're correct about the scheduler being an invisible pid 0. But it appears to work just fine as locally, so I imagine it's not a concern?

Reply:
I'm curious how it works right now then, because it appears that you send the correct ID (tracee's actual ID) if you want it to trace itself, which should actually block the tracing instead of allowing it.
Changing the `==` to `!=` in the bpf code would change that though.

Reply:
You are right, I didn't confirm it was working correctly, thanks for catching that.

Reply:
Yep, I would use MAX_PID value instead of a logic with "0", because zero might also mean the "host_tid" value inside eBPF program is not initialized (yet), and lose the last PID tracing.

IMO: DO NOT overflow the unsigned int with -1 or MAX_PID+1, that is implementation dependent and might not work as expected.

Reply:
Nah, I think it is fine to use 0 for the eBPF logic... so I'm +1 here.